storage/modules/block_storage.go

```diff
@@ -970,6 +992,19 @@ func (b *BlockStorage) storeTransaction(
 	return storeUniqueKey(ctx, transaction, hashKey, encodedResult, true)
 }
 
+func (b *BlockStorage) getBackwardRelationKeys(tx *types.Transaction) [][]byte {
+	var keys [][]byte
+	for _, relatedTx := range tx.RelatedTransactions {
```

Comment:
todo: skip if on another network

Reply:
if I recall correctly, `nil` NetworkIdentifier implies same network so only skip if not nil and network identifier != network identifier we are syncing